embedded-graphics/src/primitives/line/mod.rs

```diff
@@ -78,6 +79,48 @@ impl Line {
 
         Line::new(self.start, self.start + delta)
     }
+
+    /// Get two lines representing the left and right edges of the thick line.
+    ///
+    /// If a thickness of `0` is given, the lines returned will lie on the same points as `self`.
+    fn extents(&self, thickness: u32) -> (Line, Line) {
+        let mut it = ParallelsIterator::new(self, thickness.saturating_cast());
+        let reduce =
+            it.parallel_parameters.position_step.major + it.parallel_parameters.position_step.minor;
+
+        let mut left = (self.start, 0);
+        let mut right = (self.start, 0);
+
+        loop {
+            if let Some((bresenham, reduce)) = it.next() {
+                right = (bresenham.point, reduce);
+            } else {
+                break;
+            }
+
+            if let Some((bresenham, reduce)) = it.next() {
+                left = (bresenham.point, reduce);
+            } else {
+                break;
+            }
+        }
+
+        let delta = self.end - self.start;
+
+        let left_line = if left.1 == 0 {
```

Comment:
Should we change the `Item` type of the `ParallelsIterator`? This code only works for `0` and `1` and it's strange to use an `u32` for that purpose. I would suggest to add an `enum ParallelLineType { Normal, Extra }` and use that instead of the `u32`.

Reply:
It is a bit strange, but it's not causing any issues at the moment. Maybe change this in a future PR if it becomes more of a papercut?

Reply:
IMO this change belongs into this PR, because it's the first instance where a length reduction > 1 wouldn't be handled correctly. But if you don't want to dig into the other code at the moment you can leave it and I will create a PR for the change.

Reply:
Good point. I've added a new commit to make the change in this PR.